src/TSHelper.ts

```diff
@@ -314,7 +316,7 @@ export class TSHelper {
         return undefined;
     }
 
-    public static isBinaryAssignmentToken(token: ts.SyntaxKind): [boolean, ts.BinaryOperator] {
+    public static isBinaryAssignmentToken(token: ts.SyntaxKind): [true, ts.BinaryOperator] | [false, never] {
```

Comment:
TS simplifies `ts.BinaryOperator | never` to `ts.BinaryOperator`, which isn't exactly what's desired here.